File: src/main/java/io/github/divyeshmangla/timetable/model/Subject.java
package io.github.divyeshmangla.timetable.model;

/**
 * If no abbr is provided in the config, subject name is used as the abbreviation.
 */
public record Subject(
        String code,
        String name,
        String abbr
) {
    public Subject {
        if (abbr == null || abbr.isBlank()) {
            abbr = name;
        }
    }
}

File: internal/model/subject.go
package model

type Subject struct {
	Code string
	Name string
	Abbr string
}

func NewSubject(code, name, abbr string) Subject {
	if abbr == "" {
		abbr = name
	}

	return Subject{
		Code: code,
		Name: name,
		Abbr: abbr,
	}
}
